### ISSUE TITLE
unable to send metrics to AppDynamics

### ISSUE THREAD

**Issue by iftakharGit** [spring-cloud/spring-cloud-stream-binder-aws-kinesis#204]
**In what version(s) of Spring Cloud Stream Binder for AWS Kinesis are you seeing this issue?**

2.2.0

**Describe the bug**

I am using spring boot application to stream AWS Kinesis. I am trying to send metrics to AppDynamics such number read and writes. I am also sending metrics for AWS Managed Kafka which is able to send metrics to APPD.

**Expected behavior**

It should send metrics to AppDynamics

Also pls confirm if there is already working solution for above issue

**Comment by artembilan** [comment]
I don't think it has to do anything with this project which is just a Java client to AWS Kinesis service.
It sounds like that AppDynamics is a part of cloud configuration where your Kinesis instance is managed.

If there is an API for Kinesis Java client which is able to produce those metrics, then you are free to override a `KinesisAsyncClient` bean we auto-configure with this library.

Let us know, please, what in your opinion should be fixed exactly in this project to make it working.

NOTE: the latest version of this library is: https://github.com/spring-cloud/spring-cloud-stream-binder-aws-kinesis/releases/tag/v4.0.0.

**Comment by artembilan** [comment]
Closed as non-relevant to the project.
